pydax/_dataset.py

```diff
@@ -161,6 +161,36 @@ def is_downloaded(self) -> bool:
         return self._data_dir.is_dir() and len(os.listdir(self._data_dir)) > 0
 
 
+def handle_name_version_params(func: Callable) -> Callable:
+    """Decorator for handling ``name`` and ``version`` parameters."""
+    def wrapper_decorator(name: str,
+                          version: Optional[str] = None,
+                          *args: Union[bool, Iterable[str], None],
+                          **kwargs: Union[bool, Iterable[str], None]) -> Union[None, Dict[str, Any], SchemaDict]:
```

Comment:
Because this seems generic, perhaps consider using some `Any` here? It might make sense to use [`@no_type_check_decorator`](https://www.python.org/dev/peps/pep-0484/#compatibility-with-other-uses-of-function-annotations) as this is really meant to be a dynamic part and shouldn't enforce type checking.

Reply:
thanks good idea